g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java

```diff
@@ -91,9 +97,159 @@ public APIResponseDTO getMultTrans(String productName, String version,String pse
             translationDTO.setPseudo(new Boolean(pseudo));
         }
         translationDTO = multipleComponentsService.getMultiComponentsTranslation(translationDTO);
-        return super.handleResponse(APIResponseStatus.OK, translationDTO);
+        return translationDTO;
     }
+    
+    
+    public APIResponseDTO getMultTrans(String productName, String version, String componentsStr, String localesStr, String pseudo,
+            HttpServletRequest req)  throws Exception {
+        if(StringUtils.isEmpty(componentsStr) && StringUtils.isEmpty(localesStr)) {
+            return  super.handleResponse(APIResponseStatus.OK, getAllCompTrans( productName,  version, pseudo, req)) ;
+        }else {
+            return getV2MultipleComponentsTrans(productName, componentsStr,  version,  localesStr,  pseudo, req);
+        }
+        
+        
+    }
+   
+    
+     private TranslationDTO getPartTranslationReqParams(String productName,
+             String components, String version, String locales, String pseudo,
+             HttpServletRequest req) throws Exception {
+         TranslationDTO translationDTO = new TranslationDTO();
+         translationDTO.setProductName(productName);
+         translationDTO.setVersion(version);
+         List<String> componentList = null;
+         if (StringUtils.isEmpty(components)) {
+             componentList = productService.getComponentNameList(productName, version);
+         }else {
+             componentList = new ArrayList<String>();
+             for (String component : components.split(",")) {
+                 componentList.add(component.trim());
+             }
+         }
+         translationDTO.setComponents(componentList);
+         List<String> localeList = new ArrayList<String>();
+         if (new Boolean(pseudo)) {
+             localeList.add(ConstantsKeys.LATEST);
+         } else if (!StringUtils.isEmpty(locales)) {
+             for (String locale : locales.split(",")) {
```

Comment:
use Arrays.asList

Reply:
this point can't use Arrays.asList， need to excute getMapingLocale(productName, version, locale.trim()) method